### ISSUE TITLE
Precompilation fails: `UndefVarError: HasBranchingCtx not defined`

### ISSUE THREAD

**Issue by adrhill** [SciML/DiffEqFlux.jl#583]
I'm getting the following error when precompiling DiffEqFlux:
```julia-repl
[ Info: Precompiling DiffEqFlux [aae7a2af-3d4f-5e19-a356-7da93b79d9d0]
ERROR: LoadError: LoadError: UndefVarError: HasBranchingCtx not defined
Stacktrace:
  [1] getproperty(x::Module, f::Symbol)
    @ Base ./Base.jl:26
  [2] top-level scope
    @ ~/.julia/packages/DiffEqFlux/W9Cwj/src/fast_layers.jl:195
  [3] include(mod::Module, _path::String)
    @ Base ./Base.jl:386
  [4] include(x::String)
    @ DiffEqFlux ~/.julia/packages/DiffEqFlux/W9Cwj/src/DiffEqFlux.jl:1
  [5] top-level scope
    @ ~/.julia/packages/DiffEqFlux/W9Cwj/src/DiffEqFlux.jl:84
  [6] include
    @ ./Base.jl:386 [inlined]
  [7] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1213
  [8] top-level scope
    @ none:1
  [9] eval
    @ ./boot.jl:360 [inlined]
 [10] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [11] top-level scope
    @ none:1
in expression starting at /Users/funks/.julia/packages/DiffEqFlux/W9Cwj/src/fast_layers.jl:195
in expression starting at /Users/funks/.julia/packages/DiffEqFlux/W9Cwj/src/DiffEqFlux.jl:1
ERROR: Failed to precompile DiffEqFlux [aae7a2af-3d4f-5e19-a356-7da93b79d9d0] to /Users/funks/.julia/compiled/v1.6/DiffEqFlux/jl_J1SwhF.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::Base.TTY, internal_stdout::Base.TTY)
   @ Base ./loading.jl:1360
 [3] compilecache(pkg::Base.PkgId, path::String)
   @ Base ./loading.jl:1306
 [4] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1021
 [5] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:914
 [6] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:901
```

I'm using the following package versions:
```Julia-repl
  [2b5f629d] DiffEqBase v6.66.0
  [aae7a2af] DiffEqFlux v1.40.0
  [41bf760c] DiffEqSensitivity v6.50.2
  [0c46a032] DifferentialEquations v6.17.1
  [587475ba] Flux v0.12.1
  ```

**Comment by ChrisRackauckas** [comment]
This should be fixed now if you update. It was a v1.5 issue that showed up due to some compiler limitations.

**Comment by korsbo** [comment]
I'm still getting this with `DiffEqFlux` `v1.41.0`.

**Comment by korsbo** [comment]
Downgrading to `v1.40.1` did not help.

**Comment by ChrisRackauckas** [comment]
What version of DiffEqSensitivity? It should require

https://github.com/SciML/DiffEqFlux.jl/blob/master/Project.toml#L39

6.52 which defines it for sure.

**Comment by korsbo** [comment]
I was using an older version than that. I had some trouble with upgrading DiffEqSensitivity since it led to incompatible versions with some other package. That'll get resolved in due time but it'd be good to have compat bounds to ensure that if one can't get DiffEqSensitivity 6.52 then you'll keep using an older (and functional) version of DiffEqFlux. 

My immediate problem was solved by pinning DiffEqFlux to 1.39.0.
